src/main/java/com/server/Request.java

```diff
@@ -0,0 +1,101 @@
+package com.server;
+
+import com.server.interfaces.RequestInterface;
+import com.server.wrappers.InputStreamWrapper;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.StringTokenizer;
+
+public class Request {
+    private String method;
+    private String path;
+    private Map<String, String> headers = new HashMap<>();
+    private Map<String, String> queryParameters = new HashMap<>();
+    private BufferedReader in;
+
+    public Request(BufferedReader in)  {
+        this.in = in;
+    }
+
+    String getMethod()  {
+        return method;
+    }
+
+    String getPath()  {
+        return path;
+    }
+
+    private void parseQueryParameters(String queryString)  {
+        for (String parameter : queryString.split("&"))  {
+            int separator = parameter.indexOf('=');
+            if (separator > -1)  {
+                queryParameters.put(parameter.substring(0, separator),
+                        parameter.substring(separator + 1));
+            } else  {
+                queryParameters.put(parameter, null);
+            }
+        }
+    }
+
+    boolean parse() throws IOException {
+        String initialLine = in.readLine();
+        log(initialLine);
+        StringTokenizer token = new StringTokenizer(initialLine);
+        String[] components = new String[3];
+        for (int i = 0; i < components.length; i++)  {
+            if (token.hasMoreTokens())  {
+                components[i] = token.nextToken();
+            } else  {
+                return false;
+            }
+        }
+
+        method = components[0];
+
+
+        while (true)  {
```

Comment:
instead of doing a possible infinite loop here, can we figure out a way to find an exact amount of iteration we need to read the header?